src/librustc_lint/unused.rs

```diff
@@ -103,6 +103,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedResults {
                         cx.span_lint(UNUSED_MUST_USE, expr.span, msg);
                         op_warned = true;
                     },
+                    hir::BiAdd | hir::BiSub | hir::BiDiv | hir::BiMul | hir::BiRem => {
+                        let msg = "unused arithmetic operation which must be used";
```

Comment:
1. While you're here, maybe add the bitops too?  `a^b;` giving a warning also makes sense to me
2. Unary ops too? `-x` seems to also fit "unused arithmetic operation which must be used"
3. At that point, is it just all ops?  The only three left are `*`, `&&`, and `||`.  (The short-circuiting ones I could see an argument for not having the lint , since `foo() || panic!()` is possible, but...)

Reply:
I'm working on the `UnaryOp` pattern matching.  While matching on binary operators, it matches on `bin_op.node`.  However, the `UnOp` type doesn't have this field.  I've tried to find the definition of this type but I've come up with nothing so far.  Any suggestion as to what file I should check out?

Reply:
`UnOp` is an enum [defined here](https://github.com/rust-lang/rust/blob/master/src/librustc/hir/mod.rs#L1054). `BinOp` is an enum as well but it's [wrapped in a struct that gives it span information](https://github.com/rust-lang/rust/blob/master/src/librustc/hir/mod.rs#L1054), which I assume was considered unnecessary for unary ops.

All the `hir::` types are defined in `src/librust/hir.mod.rs`. For convenience all the enums have their variants reexported in this module as well.